x-pack/plugins/security_solution/common/endpoint/generate_data.test.ts

```diff
@@ -268,6 +267,29 @@ describe('data generator', () => {
       }
     };
 
+    it('sets the start and end times correctly', () => {
+      const startOfEpoch = new Date(0);
+      let startTime = new Date(timestampSafeVersion(tree.allEvents[0]) ?? startOfEpoch);
+      expect(startTime).not.toEqual(startOfEpoch);
+      let endTime = new Date(timestampSafeVersion(tree.allEvents[0]) ?? startOfEpoch);
+      expect(startTime).not.toEqual(startOfEpoch);
+
+      for (const event of tree.allEvents) {
+        const currentEventTime = new Date(timestampSafeVersion(event) ?? startOfEpoch);
+        expect(currentEventTime).not.toEqual(startOfEpoch);
+        if (currentEventTime < startTime) {
+          startTime = currentEventTime;
+        }
+
+        if (currentEventTime > endTime) {
```

Comment:
Don't know how likely it is for `currentEventTime` to equal `endTime` or `startTime`, but maybe just use an else as a catch all? 
```
	if (currentEventTime < startTime) { 
		startTime = currentEventTime 
	} else { 
	  endTime = currentEventTime
	}
```

Reply:
Good idea 👍

Reply:
actually they need to be separate `if statements` haha. That's why I just got a random test failure (also not great, I'll try to fix that too). Using the if-else will mean that `endTime` will be reset anytime `currentEventTime` is greater than or equal to `startTime`, which isn't what we want. We want `endTime` to be set anytime `currentEventTime` is greater than `endTime` otherwise we could set `endTime` to something less than the greatest event time.

Reply:
Cool, makes sense. Ignore me, I'm bad with 🕐